fitlins/workflows/base.py

```diff
@@ -150,160 +76,134 @@ def join_dict(base_dict, dict_list):
-        '[_rec-{reconstruction}][_run-{run}][_echo-{echo}]_bold_' \
-        '{type<design|corr|contrasts>}.svg'
-    ds_design = pe.MapNode(
-        BIDSDataSink(base_directory=out_dir, fixed_entities={'type': 'design'},
-                     path_patterns=image_pattern),
-        iterfield=['entities', 'in_file'],
-        run_without_submitting=True,
-        name='ds_design')
-
-    # ds_corr = pe.MapNode(
-    #     BIDSDataSink(base_directory=out_dir, fixed_entities={'type': 'corr'},
-    #                  path_patterns=image_pattern),
-    #     iterfield=['entities', 'in_file'],
-    #     run_without_submitting=True,
-    #     name='ds_corr')
-
-    # ds_l1_contrasts = pe.MapNode(
-    #     BIDSDataSink(base_directory=out_dir,
-    #                  fixed_entities={'type': 'contrasts'},
-    #                  path_patterns=image_pattern),
-    #     iterfield=['entities', 'in_file'],
-    #     run_without_submitting=True,
-    #     name='ds_l1_contrasts')
-
-    # ds_l2_contrasts = pe.MapNode(
-    #     BIDSDataSink(base_directory=out_dir,
-    #                  fixed_entities={'type': 'contrasts'},
-    #                  path_patterns=image_pattern),
-    #     iterfield=['entities', 'in_file'],
-    #     run_without_submitting=True,
-    #     name='ds_l2_contrasts')
-
-    contrast_plot_pattern = '[sub-{subject}/][ses-{session}/]' \
-        '[sub-{subject}_][ses-{session}_]task-{task}[_acq-{acquisition}]' \
-        '[_rec-{reconstruction}][_run-{run}][_echo-{echo}]_bold' \
-        '[_space-{space}]_contrast-{contrast}_ortho.png'
-    ds_l1_contrast_plots = pe.Node(
-        BIDSDataSink(base_directory=out_dir,
-                     path_patterns=contrast_plot_pattern),
-        run_without_submitting=True,
-        name='ds_l1_contrast_plots')
-
-    ds_l2_contrast_plots = pe.Node(
-        BIDSDataSink(base_directory=out_dir,
-                     path_patterns=contrast_plot_pattern),
-        run_without_submitting=True,
-        name='ds_l2_contrast_plots')
-
-    #
-    # Connections
-    #
+    models = []
+    for ix, step in enumerate(loader.run().outputs.steps):
+        # Set up elements common across levels
 
-    wf.connect([
         #
-        # Modeling
+        # Because pybids generates the entire model in one go, we will need
+        # various helper nodes to select the correct portions of the model
         #
-        (loader, select_l1_entities, [('entities', 'inlist')]),
-
-        (select_l1_entities, getter,  [('out', 'entities')]),
-
-        (loader, select_l1_contrasts, [('contrast_info', 'inlist')]),
 
-        (loader, l1_model, [('session_info', 'session_info')]),
-        (getter, l1_model, [('bold_files', 'bold_file'),
-                            ('mask_files', 'mask_file')]),
-        (select_l1_contrasts, l1_model,  [('out', 'contrast_info')]),
+        select_entities = pe.Node(
+            niu.Select(index=ix), name='select_{}_entities'.format(step))
```

Comment:
I believe it's theoretically possible for the same level to occur multiple times in a BIDS model.
```suggestion
            niu.Select(index=ix), name='select_l{:d}_entities'.format(ix + 1))
```